include/solvers/navier_stokes_scratch_data.h

```diff
@@ -1268,6 +1301,68 @@ class NavierStokesScratchData : public PhysicsScratchDataBase
   void
   calculate_physical_properties();
 
+  /**
+   * @brief Calculates the particle forces and velocities that were projected on
+   * the fluid dofs at the quadrature points of the velocity and pressure FE.
+   * The values are stored in the corresponding variables in scratch data.
+   *
+   * @param[in] particle_drag_cell Iterator pointing to the current active cell
+   * using the particle drag force DoFHandler
+   *
+   * @param[in] particle_two_way_coupling_force Iterator pointing to the current
+   * active cell using the particle two-way coupling force DoFHandler
+   *
+   * @param[in] particle_velocity_cell Iterator pointing to the current active
+   * cell using the particle velocity DoFHandler
+   *
+   * @param[in] particle_fluid_drag Object containing the projection of the
+   * particle drag onto the fluid dofs
+   *
+   * @param[in] particle_two_way_coupling_force Object containing the projection
+   * of the two-way coupling force projected from the particles onto the fluid
+   * dofs
+   *
+   * @param[in] particle_velocity Object containing the projection of the
+   * particle velocities onto the fluid dofs
+   */
+  // beta will be added here for the implicit/semi-implicit coupling
+  template <typename VectorType>
+  void
+  calculate_particle_fields_values(
+    const typename DoFHandler<dim>::active_cell_iterator &particle_drag_cell,
+    const typename DoFHandler<dim>::active_cell_iterator
+      &particle_two_way_coupling_force_cell,
+    const typename DoFHandler<dim>::active_cell_iterator
+                                   &particle_velocity_cell,
+    const VectorType               &particle_fluid_drag,
+    const VectorType               &particle_fluid_force_two_way_coupling,
+    const VectorType               &particle_velocity,
+    const Parameters::DragCoupling &drag_coupling)
+  {
+    constexpr FEValuesExtractors::Vector vector_index(0);
+
+    this->fe_values_particle_two_way_coupling_force->reinit(
+      particle_two_way_coupling_force_cell);
+    (*this->fe_values_particle_two_way_coupling_force)[vector_index]
+      .get_function_values(particle_fluid_force_two_way_coupling,
+                           this->particle_two_way_coupling_force_values);
+
+    if (drag_coupling == Parameters::DragCoupling::fully_explicit)
+      { // particle_drag_values will remain zero in the implicit and
+        // semi-implicit
+        // coupling, since the momentum transfer coefficient is used instead
```

Comment:
```suggestion
      { // particle_drag_values will remain zero in the implicit and
        // semi-implicit coupling, since the momentum transfer 
        // coefficient is used instead
```